### PR TITLE
S6R9

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
Fixes for infinite fuel till first gate issues

### CODE REVIEW

**Comment by BrettRyland** [comment]
I'm seeing a lot of merge conflicts with the dev branch. It looks like these changes are based off of the master branch instead with the kinetic-missile-refactor PR added in. I guess this is in order to minimise the changes from the original DLL for this round, but since the DLL is already being distributed, I think it's fine to merge in the rest of the changes from dev.

I think the kinetic-missile-refactor PR should be good to merge — yes/no? So I think the best way to get this merged is to first merge that PR, then fix the conflicts between this branch and dev.

**Comment by SuicidalInsanity** [comment]
Was sort of in a rush to get the S6R9 changes done, so didn't realize I was building off the wrong branch; was intending to just use Master to keep things as similar to the 1.1.6.12.0 release. So yes, merging the kinetic missile PR first would be the way to go

**Comment by BrettRyland** [comment]
> Was sort of in a rush to get the S6R9 changes done, so didn't realize I was building off the wrong branch; was intending to just use Master to keep things as similar to the 1.1.6.12.0 release. So yes, merging the kinetic missile PR first would be the way to go

OK, the kinetic-missile-refactor branch is merged in now, so resolving the conflicts between this and dev can happen when you have some spare time.

File: BDArmory/Control/BDGenericAIBase.cs [hunk 1]
```diff
@@ -247,6 +247,8 @@ public void TogglePilot()
         public virtual string Name { get; } = "AI Control";
         public bool Enabled => pilotEnabled;
 
+        public bool TakingOff = true;
```
Comment (BrettRyland) [inline BDArmory/Control/BDGenericAIBase.cs:250]: This is the same as `initialTakeOff` in BDModulePilot.cs. I'd suggest replacing `initialTakeOff` with this new `TakingOff` variable.
If we ever manage to make a landing routine, then that would need to reset this to `true`.

**Pull request merged**